src/App.jsx

```diff
@@ -0,0 +1,294 @@
+import { useEffect, useState } from 'react'
+
+import './App.css'
+import { constants } from './app/config'
+import { getDirection, getMoveStepPositions } from './app/game'
+import { useStateValue } from './app/state/useStateValue'
+import Dice from './components/dice/Dice'
+import Grid from './components/grid/Grid'
+import Legend from './components/legend/Legend'
+import { arrayIsInArray, log, randomNumber } from './utils/main'
+
+const debugTimeInterval = false
+
+const App = () => {
+  // hooks
+  const [state, dispatch] = useStateValue()
+  const [uiDisabled, setUIDisabled] = useState(false)
+  const [isBootleTurn, setIsBottleTurn] = useState(false)
+  // TODO: this is a temporary state until figure out a solution for Burak recommendation
+  const [rollInterval, setRollInterval] = useState(null)
```

Comment:
it is better to use `useRef` to store `rollInterval` to prevent unnecessary rerender on state changing

Reply:
yes sure, another awesome recommendation :)

FIXED and a lot better in render counts